# Assignment: Statistical Testing with Scraped-Inspired Data

## Objective

In this assignment, you will act as a data analyst investigating student performance using a provided dataset. You will:

* Identify appropriate statistical tests
* Run analyses in Python
* Interpret results in context
* Justify your methodological choices

# Background Scenario

A university is studying how different factors influence student success in an introductory programming course.

They collected data on:

* Whether students used an AI study tool
* How many hours they studied per week
* Their final exam scores
* Their major
* Whether they passed the course

The university wants to understand:

1. Does the AI tool improve performance?
2. Do outcomes differ across majors?
3. Are certain groups more likely to pass?

# Dataset Description

| Column | Description |
| -------- | -------- |
|study_group   | "Control" or "AI_Tool"   |
|hours_studied   | Number of hours studied per week   |
|score   | Final exam score   |
|major   | CS, Biology, Business   |
|passed   | Yes or No   |


In [2]:
import pandas as pd
from scipy.stats import ttest_ind, chi2_contingency, f_oneway

# Load dataset and display first 5 rows
df = pd.read_csv("student_study_data.csv") # importing the right csv file

df.head() # first 5 rows
# df.info() column, type and nulls
# df.shape rows, columns

,study_group,hours_studied,score,major,passed
0,Control,3.7,84.2,Business,No
1,AI_Tool,5.6,83.2,CS,Yes
2,AI_Tool,4.2,88.6,CS,Yes
3,Control,3.2,68.5,Business,Yes
4,AI_Tool,5.0,69.7,Biology,Yes


## Step 1: Explore Data

In [3]:
#show summary statistics (refer to command from lecture)
df.describe() # from lecture: computes basic statistics to anchor understanding of the data's scale and spread
# returns count, mean, std (standard deviation, etc. single best starting point)

,hours_studied,score
count,120.000000,120.000000
mean,5.138333,76.695833
std,1.467473,9.538092
min,2.400000,47.300000
25%,4.000000,70.475000
50%,5.100000,76.650000
75%,6.225000,82.975000
max,8.600000,101.600000


## Step 2 & 3: Statistical Tests

1. Choose an appropriate statistical test for each question and explain why
2. Run t-test, chi-square, ANOVA
3. Report statistic and p-value

### Research Question: Does the AI tool improve performance?

* Statistical Test: T-Test

* This research question requires a T-Test because the outcome is a numerical outcome (final exam score), not simply a "Yes" or "No." Further, two groups means can be compared, students who passed/didn't pass and whether or not they used AI tools.  


In [4]:
####Insert code here####
from scipy.stats import ttest_ind
group1 = df[df["study_group"] == "Control"]["score"]
group2 = df[df["study_group"] == "AI_Tool"]["score"]

stat, p = ttest_ind(group1, group2)
print(f"statistic: {stat:.3f}")
print(f"p-value: {p:.4f}")


statistic: -0.815
p-value: 0.4170


### Research Question: Do outcomes differ across majors?

* Statistical Test: ANOVA

* Since 3+ groups are being analyzed (majors) and the outcome (final exam score) is a numerical value, ANOVA is the best statisitcal test to use for this question.

In [5]:
####Insert code here####
from scipy.stats import f_oneway

groups = [
    df[df["study_group"] == s]["score"]
for s in df["study_group"].unique()
]
f_stat, p = f_oneway (*groups)
print(f"F={f_stat:.3f}, p = {p:.4f}")


F=0.664, p = 0.4170


### Research Question: Are certain groups more likely to pass?

* Statisitcal Test: Chi-Square Test

* The main reason why this research questions would not be a T-Test is because the two variables at hand are categorical, not numerical. The two variables are major and passed (Yes or No)

In [7]:
####Insert code here####
from scipy.stats import chi2_contingency

# build a contingency table
table = pd.crosstab(
    df["study_group"],
    df["passed"]
)

chi2, p, dof, expected = (
    chi2_contingency(table)
)
print(f"chi2: {chi2:.3f}, p: {p:.4f}")

chi2: 0.558, p: 0.4550


## Step 4: Interpretation
1. Explain results for each test in plain language
2. State if significant
3. Discuss limitations of the test and data


RQ 1: Does the AI tool improve performance?
Since the p-value is 0.4170 (41.7%) and is much higher than the alpha level of 0.05 (5%), we fail to reject the null hypothesis. This means there's a high chance that the outcomes are due to randomness rather than the AI tool itself. Plainly, the AI tool does not significantly improve exam performance. 



RQ 2: Do outcomes differ across majors? 
Interestingly, the p-value for the ANOVA test is the same as the T-test. Just the same, we fail to reject the null hypothesis. Since the p-value is so high, there's no significant difference in outcomes across the 3 majors. The F-Statistic (F = 0.664) represents the ration between the variation between group means and the variation within the groups. Since the F-value is much lower than 1, this signifies that the means of majors are grouped closer together rahter than spread apart. The limitation of ANOVA is that it tells you whether at least one group differents, not which one. Further testing must be used.



RQ 3: Are certain groups more likely to pass?
Again, for the Chi-test the p-value was 0.4550 which is much higher than the alpha value. This means there's a 45.5% probability that the difference between groups (AI tool and controlled) is due to chance. Further, we fail to reject the null hypothesis. 




